app/src/main/java/org/piwigo/ui/main/MainViewModel.java

```diff
@@ -99,9 +91,15 @@ public void setError(Throwable th){
     public void changeAccount(Account account) {
         userManager.setSessionCookie(null);
         userManager.setSessionToken(null);
+        displayFab.set(!userManager.isGuest(account));
+        if (account == null)
+            return;
+        if (userManager.getUsername(account).equals(username.get()))
```

Comment:
does this work if we have two accounts on different servers but the same username?

Reply:
This really shouldn't be necessary - but the function is called twice. And it shows just how fragile the counting is.

Reply:
this is left to be sorted out. changeAccount is called too often, there needs to be a clearer responsibility